### PR TITLE
Add Block Height to EpochInfo

### PR DESCRIPTION
**Pull request by jeebster**
Closes: #452

## Description

**NOTES**: 
* `start_height` and `current_epoch_start_height` fields have been added following the timing scheme. If some of this information is unnecessary, please advise and I will deprecate accordingly
* protobuf compilation via `make proto-all` appears to have bumped the cosmos-sdk version dependency
* Please review logic refactoring in epoch module *abci* and *genesis* implementation and test files; I'm not too familiar with the protocol yet so I made some assumptions relative to the issue description
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: proto/osmosis/epochs/genesis.proto [hunk 1]
```diff
@@ -28,6 +27,9 @@ message EpochInfo {
     (gogoproto.moretags) = "yaml:\"current_epoch_start_time\""
   ];
   bool epoch_counting_started = 6;
+  reserved 7;
+  int64 start_height = 8;
+  int64 current_epoch_start_height = 9;
```
[the remark this answers is not in the document]
Comment (jeebster) [reply proto/osmosis/epochs/genesis.proto]: addressed in 535e15c6feaf3055187e6901bfcd2f35dcda7a2a

**Review by ValarDragon — APPROVED** [review]
Perfect, the change LGTM! 

It just occurred to me, would it be easy to make a test where you manually protobuf serialized an epoch info prior to this change, and ensured that it can still be decoded in the new logic? 

We basically need to double check that in order to not need any state migrations before it gets on mainnet. Sorry for not giving that feedback earlier.

**Comment by jeebster** [comment]
> Perfect, the change LGTM!
> 
> It just occurred to me, would it be easy to make a test where you manually protobuf serialized an epoch info prior to this change, and ensured that it can still be decoded in the new logic?
> 
> We basically need to double check that in order to not need any state migrations before it gets on mainnet. Sorry for not giving that feedback earlier.

@ValarDragon I'll take a look into what this may entail - assuming mocking generated protobuf definitions and running backwards-compatible tests. I don't think this would be ideal but a temporary solution could be to check for the existence of the new proto property definition and removing the check after migration. Also, let me know if the base branch should be changed to a different branch if this shouldn't be targeting `main`

**Comment by ValarDragon** [comment]
The test shouldn't be too complex, protobuf _should_ still allow parsing old serialized data with the new proto format, as long as no new required fields were added. (As is the case in this PR)

So it would protobuf serializing in old code, copying those bytes out, and then making a new test in this branch that tried proto deserializing those bytes and got no error

**Comment by jeebster** [comment]
@ValarDragon fbc9a1de should address legacy `EpochInfo` message compatibility

**Comment by ValarDragon** [comment]
Perfect, thanks!

**Pull request merged**